usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -9585,3 +9585,110 @@ static void free_ep11_target_for_apqn(target_t target)
     }
 }
 
+CK_RV token_specific_set_attribute_values(STDLL_TokData_t *tokdata, OBJECT *obj,
+                                          TEMPLATE *new_tmpl)
+{
+    ep11_private_data_t *ep11_data = tokdata->private_data;
+    CK_OBJECT_CLASS class;
+    size_t keyblobsize = 0;
+    CK_BYTE *keyblob;
+    DL_NODE *node;
+    CK_ATTRIBUTE *ibm_opaque_attr = NULL;
+    CK_ATTRIBUTE_PTR attributes = NULL;
+    CK_ULONG num_attributes = 0;
+    CK_ATTRIBUTE *attr;
+    CK_RV rc;
+
+    rc = template_attribute_get_ulong(obj->template, CKA_CLASS, &class);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("%s CKA_CLASS is missing\n", __func__);
+        return rc;
+    }
+
+    switch (class) {
```

Comment:
Boolean attributes seem to be supported also on public keys and change whenever changed on a public key.